frontend/src/components/forms/EditResident.tsx

```diff
@@ -0,0 +1,314 @@
+    const inputValue = e.target.value as string;
+    if (/^[a-z]{0,2}$/i.test(inputValue)) {
+      setInitials(inputValue.toUpperCase());
+      setInitialsError(false);
+    }
+  };
+
+  const handleFlagged = () => {
+    setFlagged(!flagged);
+    setMoveOutDate(dateLeft? new Date(Date.parse(dateLeft)) : undefined);
+  };
+
+  const handleRoomNumberChange = (e: { target: { value: unknown } }) => {
+    const inputValue = e.target.value as string;
+    if (inputValue !== null && /^[0-9]{0,3}$/.test(inputValue)) {
+      setRoomNumber(parseInt(inputValue, 10));
+      setRoomNumberError(false);
+    }
+  };
+
+  const handleMoveInDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (moveOutDate && inputValue > moveOutDate) {
+        setMoveInDateError(true);
+      }
+      else {
+        setMoveInDate(inputValue);
+        setMoveInDateError(false);
+      }
+    }
+  };
+  
+  const handleMoveOutDateChange = (inputValue: Date) => {
+    if (inputValue !== null) {
+      if (inputValue >= moveInDate) {
+        setMoveOutDate(inputValue);
+        setMoveOutDateError(false);
+      }
+      else {
+        setMoveOutDateError(true);
+      }
+    } 
+  };
+
+  const handleBuildingChange = (
+    selectedOption: SingleValue<{ label: string; value: string }>,
+  ) => {
+    if (selectedOption !== null) {
+      setUserBuilding(selectedOption.value);
+      setBuildingError(false);
+    }
+  };
+
+  const handleSubmit = () => {
+    setInitialsError(initials.length !== 2);
+    setRoomNumberError(roomNumber.toString().length !== 3);
+    setBuildingError(userBuilding === "");
+
+    //  Prevents form submission if any required values are incorrect
+    if (
+      initials.length !== 2 ||
+      moveInDateError || 
+      roomNumberError ||
+      userBuilding === "" ||
+      moveOutDateError
+    ) {
+      return;
+    }
+    editRes();
+    setShowAlert(true);
+  };
+
+  // Timer to remove alert
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  return (
+    <>
+      <Box>
+        <Modal isOpen={isOpen} onClose={toggleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>Edit Resident</ModalHeader>
+            <ModalCloseButton size="lg" />
+            <ModalBody>
+              <Divider />
+              <Row style={{ marginTop: "16px" }}>
+                <Col>
+                  <FormControl isRequired isInvalid={initialsError}>
+                    <FormLabel>Resident Initials</FormLabel>
+                    <Input
+                      placeholder={initial}
+                      value={initials}
+                      onChange={handleInitialsChange}
+                    />
+                    <FormErrorMessage>
+                      Resident Initials are required and must contain 2 letters.
```

Comment:
nit: I'd prefer if we moved the error messages into an enum and simply accessed them using `EditResidentError.RESIDENT_INITIALS` or something along those lines, to help declutter the jsx